### PR TITLE
ci: bump actions to latest versions

### PR DESCRIPTION
**Pull request by alcpereira**
# Summary

In CI there was a lot of deprecation warnings, a bump for each of the deprecated actions should fix it for now.

TLDR of the major bumps
- [actions/checkout](https://github.com/actions/checkout/releases/tag/v4.0.0): Node 20
- [actions/configure-pages](https://github.com/actions/configure-pages/releases): Node 20 and Next.js relating things
- [actions/upload-pages-artifact](https://github.com/actions/upload-pages-artifact/releases): New Github Artifacts API
- [actions/deploy-pages](https://github.com/actions/deploy-pages/releases): Uses  `actions/upload-pages-artifact` v3 or newer

With the new Github Artifacts API, the page URL is not needed anymore and retrieved through the Artifact id.

## How did you test this change?

Tested it on my fork [here](https://github.com/alcpereira/gh-dash/actions/runs/11714252937)

### CODE REVIEW

**Review by dlvhdr — APPROVED** [review]
ty!
